libs/net/dal/Services/UserService.cs

```diff
@@ -117,6 +117,12 @@ public IEnumerable<User> FindByEmail(string email)
             .Where(u => u.Email.ToLower() == email.ToLower());
     }
 
+    public IEnumerable<User> FindByPreferredEmail(string email)
```

Comment:
I think it makes more sense to update the FindByEmail() function to simply look in both the email and the preferred email.  The function is just looking for users with the specified email address.